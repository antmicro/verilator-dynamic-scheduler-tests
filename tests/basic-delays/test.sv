module t (/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;
   int cnt;

   logic clk_generated = 0;
   int  cyc;

   initial begin
      // NOTE: the delays at the beginning of the threads that only print some
      // output are on purpos "out of phase" with the clock generation.
      // This way the order of the $write output is guaranteed. If two $write
      // calls occured at the same simulation time, any order would be
      // acceptable
      #5;
      $write("Simple 1\n");
      #50;
      $write("Simple 1 post delay\n");
   end

   initial begin
      #75;
      $write("Simple 2\n");
      #70;
      $write("Simple 2 post delay\n");
   end

   initial begin
      forever begin
         #10 clk_generated = ~clk_generated;
      end
   end

   always @ (posedge clk_generated) begin
      cyc = cyc + 1;
      if (cyc % 2 == 0) begin
         $write("+\n");
      end else begin
         $write("-\n");
      end

      if (cyc > 20) begin
         $write("*-* All Finished *-*\n");
         $finish;
      end
   end

endmodule
