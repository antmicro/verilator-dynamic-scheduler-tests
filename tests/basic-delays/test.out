Simple 1
-
+
-
Simple 1 post delay
+
Simple 2
-
+
-
Simple 2 post delay
+
-
+
-
+
-
+
-
+
-
+
-
+
-
*-* All Finished *-*
